chatkit-core/src/main/kotlin/com/pusher/chatkit/CurrentUser.kt

```diff
@@ -233,45 +359,104 @@ class CurrentUser(
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMessage(roomId, messageText, attachment) }, callback)
 
+    /**
+     * Send a [messageText] to the [roomId]. The [callback] will be called with a [Result] when the
+     * operation is complete. If the operation was successful you will receive an [Int] which
+     * contains the message id, or an [Error] informing you of what went wrong.
+     */
     fun sendSimpleMessage(
             roomId: String,
             messageText: String,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendSimpleMessage(roomId, messageText) }, callback)
 
+    /**
+     * @see [sendSimpleMessage]
+     */
     fun sendSimpleMessage(
             room: Room,
             messageText: String,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendSimpleMessage(room, messageText) }, callback)
 
+    /**
+     * Sends some message [parts] to the [room]. For more information on what the [parts] can be check
+     * out the [docs][https://pusher.com/docs/chatkit/reference/android].
+     * The [callback] will be called with a [Result] when the operation is complete. If the
+     * operation was successful you will receive an [Int] which contains the sent message id, or an
+     * [Error] informing you of what went wrong.
+     *
+     * e.g.
+     * `sendMultipartMessage(room, listOf(`
+     * `NewPart.Inline("hello world"),`
+     * `NewPart.Url("https://example.com/uploads/myphoto.png", "image/png") ), callback = { })`
+     */
     fun sendMultipartMessage(
             room: Room,
             parts: List<NewPart>,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMultipartMessage(room, parts) }, callback)
 
+    /**
+     * @see [sendMultipartMessage]
+     */
     fun sendMultipartMessage(
             roomId: String,
             parts: List<NewPart>,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMultipartMessage(roomId, parts) }, callback)
 
+    /**
+     * Sets that the current user is typing in the [room]. You can send as many of these as you like,
+     * the SDK will handle rate limiting the requests. The [callback] will be called with a [Result]
```

Comment:
Would it be more correct to say "you can call this as frequently as you like" than "send as many as you like"?
We could add a suggested use case, like "e.g. on every keystroke".

Reply:
yes! that's what I was trying to go for :) thank you!